### ISSUE TITLE
R Package: force silence?

### ISSUE THREAD

**Issue by Laurae2** [Microsoft/LightGBM#191]
Is there a way to for silence in the R package? I'm coding a stochastic optimizer which makes use of LightGBM, and LightGBM writes so much to the interactive console that I cannot use my optimizer without silencing it (it slows down heavily over time - I would need about 1M, or even more boosting iterations scattered across 10K+ models, R interactive console can't bear 1M lines in console).

The specific conditions I would need:

* no warning/info messages (another verbosity level?)
* keep logging the metric during training (can't use verbose=0 and log metrics at the same time, I want to keep metric but keep silent also)
* not show randomly ["free booster handle"](https://github.com/Microsoft/LightGBM/blob/master/R-package/R/lgb.Predictor.R) and ["free dataset handle"](https://github.com/Microsoft/LightGBM/blob/master/R-package/R/lgb.Dataset.R) (this one is easy to fix)
* not print line breaks (providing no verbose and no validation data makes line breaks print)

This would be useful for users who tend to runs batched code and don't want to see anything other than errors.

For instance, I want all these messages to never appear on the console (nor all the line breaks appearing when verbose is 0):

```r
[1] "free booster handle"
[1] "free dataset handle"
[1] "free dataset handle"
[1] "free booster handle"
[1] "free dataset handle"
[1] "free dataset handle"
[LightGBM] [Warning] Ignoring Column_60 , only has one value
[LightGBM] [Warning] Ignoring Column_67 , only has one value
[LightGBM] [Warning] Ignoring Column_95 , only has one value
[LightGBM] [Info] Number of data: 1459, number of features: 133
[LightGBM] [Info] No further splits with positive gain, best gain: -inf, leaves: 22
[1]:	test's l2:43118 
Will train until hasn't improved in 10 rounds.

[LightGBM] [Info] No further splits with positive gain, best gain: -inf, leaves: 40
[2]:	test's l2:42209.8 
[LightGBM] [Info] No further splits with positive gain, best gain: -inf, leaves: 32
[3]:	test's l2:41768.4 
```

or such wall of line breaks:

```r
[LightGBM] [Warning] Ignoring Column_101 , only has one value
[LightGBM] [Warning] Ignoring Column_110 , only has one value
[LightGBM] [Warning] Ignoring Column_178 , only has one value
[LightGBM] [Warning] Ignoring Column_180 , only has one value
[LightGBM] [Warning] Ignoring Column_319 , only has one value
[LightGBM] [Warning] Ignoring Column_392 , only has one value
[LightGBM] [Warning] Ignoring Column_797 , only has one value
[LightGBM] [Warning] Ignoring Column_798 , only has one value
[LightGBM] [Warning] Ignoring Column_801 , only has one value
[LightGBM] [Warning] Ignoring Column_802 , only has one value
[LightGBM] [Warning] Ignoring Column_809 , only has one value
[LightGBM] [Warning] Ignoring Column_810 , only has one value
[LightGBM] [Warning] Ignoring Column_817 , only has one value
[LightGBM] [Warning] Ignoring Column_818 , only has one value
[LightGBM] [Warning] Ignoring Column_821 , only has one value
[LightGBM] [Warning] Ignoring Column_822 , only has one value
[LightGBM] [Warning] Ignoring Column_846 , only has one value
[LightGBM] [Warning] Ignoring Column_895 , only has one value
[LightGBM] [Warning] Ignoring Column_899 , only has one value
[LightGBM] [Warning] Ignoring Column_954 , only has one value
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
```

**Comment by guolinke** [comment]
@Laurae2 I separate the verbose and record now.
```verbose`` will control all message output and set ```record=TRUE``` will save eval result into booster.

**Comment by Laurae2** [comment]
@guolinke Thanks, works perfectly, it requires the three following parameters to force full silence and keep recording:

```r
verbosity = -1 # Set higher if you want messages such as "[LightGBM] [Warning] Ignoring Column_85 , only has one value"
verbose = -1 # Hides nearly all output
record = TRUE # Keep recording messages
```

This will be very useful for any batch users who wants to see only the issues/errors.
